### PR TITLE
woof 20220202

### PR DESCRIPTION
**Pull request by samford**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

This updates `woof` to the latest commit from 2021-06-09. The last release on the [first-party site](http://www.home.unix-ag.org/simon/woof.html) is from 2012-05-31 and the homepage directs users to the [GitHub repository](https://github.com/simon-budig/woof/) from 2020-06-24 onward.

Unfortunately, the repository has no tags/releases, so this PR is uses a commit archive as `stable`. With this situation, we'll simply have to manually check for new commits periodically and update the formula until/unless upstream starts tagging versions.

I've added `depends_on "python@3.10"`, as the `woof` file is a Python script with a `#!/usr/bin/env python3` hashbang. If this isn't necessary in this context, let me know and I'll remove it.

Past that, this adds `license "GPL-2.0-or-later"`, referencing the license comment in the `woof` file, which reads:

```
#  woof -- an ad-hoc single file webserver
#  Copyright (C) 2004-2009 Simon Budig  <simon@budig.de>
#
#  This program is free software; you can redistribute it and/or modify
#  it under the terms of the GNU General Public License as published by
#  the Free Software Foundation; either version 2 of the License, or
#  (at your option) any later version.
#
#  This program is distributed in the hope that it will be useful,
#  but WITHOUT ANY WARRANTY; without even the implied warranty of
#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#  GNU General Public License for more details.
#
#  A copy of the GNU General Public License is available at
#  http://www.fsf.org/licenses/gpl.txt, you can also write to the
#  Free Software  Foundation, Inc., 59 Temple Place - Suite 330,
#  Boston, MA 02111-1307, USA.
```

### CODE REVIEW

**Comment by SMillerDev** [comment]
> Unfortunately, the repository has no tags/releases, so this PR is uses a commit archive as stable. With this situation, we'll simply have to manually check for new commits periodically and update the formula until/unless upstream starts tagging versions.

If there are releases we shouldn't make ones up. Either they start tagging or we remove the formula, with 8 installs in 30 days we're not exactly talking about a big formula here.

**Review by SMillerDev — DISMISSED** [review]
See earlier comment

**Comment by samford** [comment]
The last official release (from the homepage) is 20120531 but it doesn't support Python 3. Support was added in a 2012-05-31 commit but these changes aren't included as part of the release on the same date (it uses the commit before the Python 3 changes). The related commit wasn't made public until the GitHub project was created (2020-06-24).

The latest commit is from 2021-06-09, so the project doesn't seem to be dead. I'll create an issue to see if they'll start tagging versions (e.g., `2021-06-09`).

**Comment by samford** [comment]
By request, upstream tagged a version ([`woof-20220202`](https://github.com/simon-budig/woof/releases/tag/woof-20220202), so I've updated this to use the tag archive.

With that out of the way, we just need to figure out if/how we can update the test to work on Linux.

**Comment by samford** [comment]
By request, upstream tagged a version ([`woof-20220202`](https://github.com/simon-budig/woof/releases/tag/woof-20220202)), so I've updated this to use the tag archive.

The test failures on macOS 11 and Linux appeared to be related to system `python3`. That is to say, `depends_on "python@3.10"` doesn't do anything meaningful by itself when we're installing a Python script that uses a `#!/usr/bin/env python3` shebang.

Calling `inreplace "woof", "/usr/bin/env python3", Formula["python@3.10"].opt_bin/"python3"` before `bin.install "woof"` works but is this the preferred approach for handling installed Python scripts like this?

File: Formula/woof.rb [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
 
   bottle do
     sha256 cellar: :any_skip_relocation, all: "854310e14cd958d44e42c8b6dd5aac3ca360ae2154763a0967cfbe4dee76a512"
   end
 
+  depends_on "python@3.10"
+
   def install
-    bin.install "woof-2012-05-31.py" => "woof"
+    inreplace "woof", "/usr/bin/env python3", Formula["python@3.10"].opt_bin/"python3"
```
Comment (branchvincent) [inline Formula/woof.rb:15]: We actually have a helper for this:
```suggestion
    rewrite_shebang detected_python_shebang, "woof"
```

You'll also need to `include Language::Python::Shebang`

Comment (samford) [reply Formula/woof.rb]: I had done some cursory `grep`ing to try to find relevant patterns but, unfortunately, didn't encounter one using this helper. Thanks for the heads up!

**Comment by samford** [comment]
This has been updated to use `rewrite_shebang detected_python_shebang, "woof"`, which should resolve any outstanding requests/questions. I imagine this should be good to merge if/when this passes CI.

**Review by SMillerDev — APPROVED** [review]

**Pull request closed without merging**